systemtest/benchtest/eventhandler/handler.go

```diff
@@ -0,0 +1,225 @@
+}
+
+// New creates a new tracehandler.Handler
+func New(p string, t transport.Transport, storage fs.FS, warmup uint) (*Handler, error) {
+	r, err := regexp.Compile(p)
+	if err != nil {
+		return nil, err
+	}
+
+	var buf bytes.Buffer
+	zw, err := zlib.NewWriterLevel(&buf, zlib.BestCompression)
+	if err != nil {
+		return nil, err
+	}
+	writer := compressedWriter{
+		buf:     &buf,
+		zwriter: zw,
+	}
+	scannerBuf := make([]byte, maxScanTokenSize)
+
+	h := Handler{
+		transport:    t,
+		warmupEvents: warmup,
+	}
+	err = fs.WalkDir(storage, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() || !r.MatchString(path) {
+			return nil
+		}
+
+		f, err := storage.Open(path)
+		if err != nil {
+			return err
+		}
+		s := bufio.NewScanner(f)
+		s.Buffer(scannerBuf, 0)
+		var wroteMeta bool
+		var scanned uint
+		for s.Scan() {
+			line := s.Bytes()
+			if len(line) == 0 {
+				continue
+			}
+			// TODO(marclop): Suppport RUM headers and handle them differently.
+			if bytes.HasPrefix(line, rumMetaHeader) {
+				return errors.New("rum data support not implemented")
+			}
+
+			if !wroteMeta {
+				writer.WriteLine(line)
+				wroteMeta = true
+				continue
+			}
+			isMeta := bytes.HasPrefix(line, metaHeader)
+			if !isMeta {
+				writer.WriteLine(line)
+				scanned++
+				continue
+			}
+
+			// Since the current token is a metadata line, it means that we've
+			// read the start of the next batch, because of that, we'll flush,
+			// close and copy the compressed contents. After we've ensured that
+			// we've written the whole batch to the buffer, we reset it and the
+			// zlib.Writer so it can be used for the next iteration.
+			if err := writer.FlushClose(); err != nil {
+				return err
+			}
+			h.batches = append(h.batches, batch{
+				r:     bytes.NewReader(writer.Bytes()),
+				items: scanned,
+			})
+			writer.Reset()
+			writer.WriteLine(line)
+			scanned = 0
+		}
+
+		if err := writer.FlushClose(); err != nil {
+			return err
+		}
+		h.batches = append(h.batches, batch{
+			r:     bytes.NewReader(writer.Bytes()),
+			items: scanned,
+		})
+		buf.Reset()
+		writer.Reset()
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	if len(h.batches) == 0 {
+		return nil, errors.New("eventhandler: regex matched no files, please specify a valid regex")
+	}
+	return &h, nil
+}
+
+// SendBatches sends the loaded trace data to the configured transport. Returns
+// the total number of documents sent and any transport errors.
+func (h *Handler) SendBatches(ctx context.Context) (uint, error) {
+	var sentEvents uint
+	sendEvents := func(r io.ReadSeeker, events uint) error {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			defer r.Seek(0, io.SeekStart)
+		}
+		// TODO(marclop) by reusing the Go agent's transport, we're not able to
+		// set the user agent accordingly. We will need to identify when we have
+		// RUM data and use a different method for sending the traces.
```

Comment:
IMO it's not worth using the Go agent's transport at all. All it's doing is setting the HTTP headers, which we can easily do here like we do in https://github.com/elastic/apm-server/blob/4127d453e5f083911c3638fc41e0337285862224/systemtest/intake.go#L50-L54

We would also need to set the Authorization and Content-Encoding headers.